test/collections.js

```diff
@@ -9,278 +9,313 @@ const { expect } = chai;
+    testnetCollectionsData.map((col, index) => {
+      const { jobIDs } = col;
+      const isValid = jobIDs.every((ele) => typeof ele === "number");
+      expect(isValid).to.be.eq(
+        true,
+        `Testnet Collection[${index}] jobIDs should be number`
+      );
+    });
+    mainnetCollectionsData.map((col, index) => {
+      const { jobIDs } = col;
+      const isValid = jobIDs.every((ele) => typeof ele === "number");
+      expect(isValid).to.be.eq(
+        true,
+        `Mainnet Collection[${index}] jobIDs should be number`
+      );
     });
+  });
+
+  it("Collection jobIDs should send OK response ", async () => {
+    for (const col of testnetCollectionsData) {
+      const { jobIDs } = col;
+      for (const ele of jobIDs) {
+        const { url, name } = testnetJobsData[ele - 1];
+        if (url.startsWith("{")) {
+          const res = await fetchCustomURL(url, name);
+          expect(res.status).to.be.equal(
+            200,
+            `Testnet Job(${name}) url should send OK response`
+          );
+        } else {
+          const res = await chai.request(url).get("").timeout(10000);
+          expect(res.status).to.be.equal(
+            200,
+            `Testnet Job(${name}) url should send OK response`
+          );
+        }
+
+        await sleep(SLEEP_TIME);
+      }
+    }
+
+    for (const col of mainnetCollectionsData) {
+      const { jobIDs } = col;
+      for (const ele of jobIDs) {
+        const { url, name } = mainnetJobsData[ele - 1];
+        if (url.startsWith("{")) {
+          const res = await fetchCustomURL(url, name);
+          expect(res.status).to.be.equal(
+            200,
+            `Mainnet Job(${name}) url should send OK response`
+          );
+        } else {
+          const res = await chai.request(url).get("").timeout(10000);
+          expect(res.status).to.be.equal(
+            200,
+            `Mainnet Job(${name}) url should send OK response`
+          );
+        }
+        await sleep(SLEEP_TIME);
+      }
+    }
+  }).timeout(4 * 60 * 1000);
 
-    it("Collection jobIDs should be number", () => {
-        testnetCollectionsData.map((col, index) => {
-            const { jobIDs } = col;
-            const isValid = jobIDs.every((ele) => typeof ele === "number");
-            expect(isValid).to.be.eq(
-                true,
-                `Testnet Collection[${index}] jobIDs should be number`
+  it("Collection JSON jobIDs should have required selectors ", async () => {
+    for (const col of testnetCollectionsData) {
+      const { jobIDs } = col;
+      for (const ele of jobIDs) {
+        const { url, name, selectorType, selector } = testnetJobsData[ele - 1];
+
+        //JSON
+        if (selectorType === 0) {
+          if (url.startsWith("{")) {
+            const { returnType } = parseUrl(url);
+            const res = await fetchCustomURL(url, name);
+            const result = decodeUniswapV2Data(res.data.result, returnType);
```

Comment:
@rajkharvar can you make this more general, to support any type of post job ie, UniswapV3, pro apis etc.

Reply:
Make sense. Resolved in [d078276](https://github.com/razor-network/datasources/pull/86/commits/d07827601a9e5259bef4c6a3f47d7afb8b3dcc0b)